### PR TITLE
Add degrees, e, and sha512 Presto functions

### PR DESCRIPTION
**Pull request by pramodsatya**
Add three Presto math functions:
degrees(x) → double (https://prestodb.io/docs/current/functions/math.html#degrees)
Converts angle x in radians to degrees.
e() → double (https://prestodb.io/docs/current/functions/math.html#e)
Returns the constant Euler’s number.
sha512(binary) → varbinary (https://prestodb.io/docs/current/functions/binary.html?highlight=sha#sha512)
Computes the sha512 hash of binary.

#1697

### CODE REVIEW

**Comment by pramodsatya** [comment]
> @pramodsatya Nice PR. Thank you for adding more Presto functions. Would you add documentation for these functions to velox/docs/functions/math.rst?
> 
> The convention is to use "Add degrees, e, and sha512 Presto functions" for the PR title. Would you update?
> 
> In comments, PR title and description, use active verbs. E.g. "Following scalar functions added to Velox:" -> "Add three Presto math functions:"

Sure, I will make these changes, thanks for the feedback

File: velox/functions/prestosql/tests/StringFunctionsTest.cpp [hunk 1]
```diff
@@ -1053,6 +1053,28 @@ TEST_F(StringFunctionsTest, sha256) {
   EXPECT_EQ(std::nullopt, sha256(std::nullopt));
 }
 
+TEST_F(StringFunctionsTest, sha512) {
+  const auto sha512 = [&](std::optional<std::string> arg) {
+    return evaluateOnce<std::string, std::string>(
+        "sha512(c0)", {arg}, {VARBINARY()});
+  };
+
+  EXPECT_EQ(
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/StringFunctionsTest.cpp:1062]: Might be good to add a test for sha512 of an empty string also.

File: velox/functions/prestosql/tests/ArithmeticTest.cpp [hunk 2]
```diff
@@ -399,6 +399,21 @@ TEST_F(ArithmeticTest, radians) {
   EXPECT_DOUBLE_EQ(-1.0000736613927508, radians(-57.3).value());
 }
 
+TEST_F(ArithmeticTest, degrees) {
+  const auto degrees = [&](std::optional<double> a) {
+    return evaluateOnce<double>("degrees(c0)", a);
+  };
+
+  EXPECT_EQ(std::nullopt, degrees(std::nullopt));
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ArithmeticTest.cpp:407]: Please add tests for values std::numeric_limits<double>::min(), max(), infinity(), quiet_NaN()

**Review by aditi-pandit — COMMENTED** [review]
Thanks Pramod for these changes. They look good.

**Comment by majetideepak** [comment]
@pramodsatya You have included some new submodules. Can you remove them?

**Review by mbasmanova — COMMENTED** [review]
@pramodsatya Looks good % small comment and format-check CI job failing. The PR is in draft mode. Is this accurate?

File: velox/functions/prestosql/StringFunctions.h [hunk 3]
```diff
@@ -93,6 +93,17 @@ struct Sha256Function {
   }
 };
 
+/// sha512(varbinary) -> varbinary
+template <typename T>
+struct Sha512Function {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  template <typename TTo, typename TFrom>
+  FOLLY_ALWAYS_INLINE bool call(TTo& result, const TFrom& input) {
```
[the remark this answers is not in the document]
Comment (pramodsatya) [reply velox/functions/prestosql/StringFunctions.h:102]: Thanks for the comment, I will check this further, I will move it from draft mode to review after fixing the format check today

**Comment by majetideepak** [comment]
@mbasmanova any other feedback for this? Thanks.

**Comment by mbasmanova** [comment]
@pramodsatya Pramod, would you rebase so I could merge this?

**Pull request closed without merging**